router/auth.go

```diff
@@ -84,6 +84,42 @@ func authPassThrough(log *log.Logger, w http.ResponseWriter, req *http.Request,
 
 func authInjectECS(log *log.Logger, w http.ResponseWriter, req *http.Request, target *targetPassThroughAuth) bool {
 	req.SetBasicAuth(target.config.Username, target.config.Password)
+	var didWhat string
+	var toWhat string
+	actionUrl := strings.Split(req.URL.String(), "/") // Example: "ecs/ACCESS/FWVERSION/" or "ecs/sam/ForceSim1/"
+	if len(actionUrl) == 5 {
+		switch actionUrl[2] {
+		case "ACCESS":
+			didWhat = actionUrl[3]
+			toWhat = "site gate: " + actionUrl[4]
+		case "sam":
+			didWhat = actionUrl[3]
+			toWhat = "site: " + actionUrl[4]
+		default:
+			didWhat = req.URL.String()
+			toWhat = req.URL.String()
+		}
+	} else {
+		didWhat = req.URL.String()
+		toWhat = req.URL.String()
+	}
+	b := new(bytes.Buffer)
+	b.WriteString(`{"did_what":"` + didWhat + `", "to_what":"` + toWhat + `"}`)
+	logActionReq, err := http.NewRequest("POST", "http://localhost:2016/logaction", b)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return false
+	}
+	for _, c := range req.Cookies() {
+		logActionReq.AddCookie(c)
+	}
+	logActionResp, err := http.DefaultClient.Do(logActionReq)
+	if err != nil {
+		log.Errorf("Error logging user action", err)
+		http.Error(w, err.Error(), http.StatusGatewayTimeout)
+		return false
+	}
+	defer logActionResp.Body.Close()
```

Comment:
Thumbs up for remembering this, there is however no need for the `defer` here.  
It is the last statement before the `return true` statement which is the final code return path for the function (there are thus no other code paths that return the function call after the `defer` is requested)